### PR TITLE
 feat: show movie revenue on register screen, use TMDB for person lookup by default and fallback to IMDB if needed

### LINKED ISSUE #56: show revenue per movie on register screen
## Prerequisites

Requires these to be implemented first: #57, #55

## Scope

- when showing list of movies on /Register screen (as implemented in #55), display a collected revenue per movie per region (as implemented in #57)

## Roles

gardener: @troggy 20 DAI
worker: @troggy 
reviewer: [reviewer]

### LINKED ISSUE #64: Cannot get movies for users without imdb_id in TMDB
When we store users (RH) right now, we store their imdb id along the name. We use this id to fetch the RH's movies then.

The problem is that themoviedb.org API doesn't return any titles for users which have no imdb_id in their database (e.g. Alan). Alan has imdb id, but themoviedborg erroneously miss it and thus the movie list for Alan is empty. It is not hard to find other people with the data missing (e.g. https://www.imdb.com/name/nm8602899/?ref_=nv_sr_srsg_0)

Solution: make sure we always have tmdbId (id in themoviedb.org) for all the users as well as IMDB ID. When fetching movies use TMDB ID instead of imdb — it seem to work and we can get Alan's (and other's) movies this way.

## Scope
BE:
- store tmdbId for User record as well as imdb_id

FE:
On /Register screen
- use TMDB to get the list of suggestions on the name
- once selected get the record for the selected person from TMDB, check if it has IMDB_ID.
- if it has IMDB, proceed as usual (store both IDs, name and email)
- if it has no IMDB id, fallback to fullname seach using unofficial-and-probably-illegal IMDB Suggestions API, get the first match and ask user to confirm the "known for" entry. Once user confirms we can safely use IMDB id from Suggestion as an IMDB ID for the TMDB person we found before.

## Roles

gardener: @troggy 50 DAI
worker: @troggy 
reviewer: [reviewer]

### PR DESCRIPTION
**Pull request by troggy**
Please remember the [Contributing Guidelines](https://WhiteRabbitTheWorldIsYours/dashboard/blob/master/CONTRIBUTION.md) :heart:

## What does this PR do?

- show movie revenue on movie screen
- change `addUser` to require `TMDB id` field
- use `TMDB id` to get movie data
- change /Register to fallbakc to IMDB suggestion if there is no IMDB id in themoviedb.org

## Recommendations for testing
- example of person without imdb_id in themoviedb.org is "Alan R. Milligan"
- person with some revenue "Daniel Baur" (producer of Paterson which we have payment for already)

## Links to relevant issues or information
Closes #64 
Closes #56

### CODE REVIEW

File: frontend/src/stores/theGraph.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { RevenuePerMovie } from './API';
+
+const THE_GRAPH_API = 'https://api.thegraph.com/subgraphs/name/clearwood/white-rabbit-cp';
```
Comment (mwaa) [inline frontend/src/stores/theGraph.ts:3]: @troggy It think needs to change to reflect new endpoints

https://api.thegraph.com/subgraphs/name/whiterabbittheworldisyours/collectionpool

**Comment by Clearwood** [comment]
image
@troggy I am getting typescript errors from @whiterabbitjs/dashboard-common

**Comment by troggy** [comment]
you might need `npm link @whiterabbitjs/dashboard-common` in both `backend` and `frontend` for this to work locally

**Comment by Clearwood** [comment]
ah that was a hazzle getting the symlinks to work, npm link is annoying.
Is it expected behavior that only those who are on IMDB can register?
What about collecting agents?

**Comment by Clearwood** [comment]
image

**Comment by troggy** [comment]
> Is it expected behavior that only those who are on IMDB can register?

Yes, Alan approved that assumption bazilion times :)

**Comment by troggy** [comment]
@Clearwood more context for the error on screen please. Mind reading is not working today :) Which user was it?

**Comment by troggy** [comment]
ok, mind reading works.. I guess this was IMDB Event which has `first_air_date` instead of `release_date`. Added a guard

**Comment by Clearwood** [comment]
oh sorry for not providing enough context, I am taking a lot with the review today so my comments were not yet published.

File: frontend/src/containers/components/MovieListWithRevenue.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    render: (url: string, movie: TMDBMovie) => <img src={`https://image.tmdb.org/t/p/w500${url}`} height={50} alt={movie.title} />,
+  },
+  {
+    title: '',
+    dataIndex: 'title',
+    key: 'title',
+    render: (title: string, movie: TMDBMovie) => (
+      <div>
+        {title}
+        {movie.release_date && (
```
Comment (Clearwood) [inline frontend/src/containers/components/MovieListWithRevenue.tsx:28]: I am still getting errors for Adam Driver

Comment (Clearwood) [reply frontend/src/containers/components/MovieListWithRevenue.tsx:28]: never mind, that seemed to have been some kind of weird caching

File: frontend/src/containers/components/PersonSearch.tsx [hunk 3]
```diff
@@ -10,49 +12,115 @@ type PersonSearchProps = {
...
+
+  const currentSuggestion = imdbSuggestions ? imdbSuggestions[imdbSuggestionIndex] : null;
   return (
-    <Select
-      showSearch
-      placeholder={placeholder}
-      optionLabelProp="value"
-      onSearch={onSearch}
-      onChange={triggerChange}
+    <Popconfirm
```
Comment (Clearwood) [inline frontend/src/containers/components/PersonSearch.tsx:79]: When is this triggered exactly?

Comment (troggy) [reply frontend/src/containers/components/PersonSearch.tsx:79]: on click. Becomes visible on demand (see `visible` parameter)

Comment (Clearwood) [reply frontend/src/containers/components/PersonSearch.tsx:79]: how can I test it?

Comment (troggy) [reply frontend/src/containers/components/PersonSearch.tsx:79]: use "Alan R. Milligan" as the name

Comment (Clearwood) [reply frontend/src/containers/components/PersonSearch.tsx:79]: thx :) looks great. one thing that caught my attention:
If I change the after a suggestion appears the suggestion is not removed. 
image

**Pull request merged**